### ISSUE TITLE
HttpClient doesn't configure the Spring RestTemplate properly.

### ISSUE THREAD

**Issue by hdva2502** [SAP/cloud-sdk-java#659]
### Describe the Bug

Hi colleagues,

when use the 
ApacheHttpClient5Accessor.getHttpClient(destination)
feature the retrieve a httpclient from an http destination (internet and onPremise).

Then we use the retrieved http client to create a Spring RestTemplate like this
new RestTemplate(new HttpComponentsClientHttpRequestFactory(httpClient));

This RestTemplate is then used for API calls, also http HEAD to the backend to check connectivity.
See also the issues you recently resolved: [#632](https://github.com/SAP/cloud-sdk-java/issues/632), [#602](https://github.com/SAP/cloud-sdk-java/issues/602).

The http client is configured correctly (see first screenshot), however the request is not configured corretly (see second screenshot)

If you look deeper into HttpComponentsClientHttpRequestFactory => [createRequest](https://github.com/spring-projects/spring-framework/blob/ac5c8adb9830939e2329f1e16727c522a172c7c8/spring-web/src/main/java/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.java#L209) you see that the relevant method is [createRequestConfig](https://github.com/spring-projects/spring-framework/blob/ac5c8adb9830939e2329f1e16727c522a172c7c8/spring-web/src/main/java/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.java#L250) where your HttpClientWrapper is not used for configuration ... because it doesn't implement Configurable ... however the HttpClient you wrap does.

So imho it would help if ApacheHttpClient5Wrapper would implement Configurable and would delegate the getConfig to the inner httpClient.

What do you think?

Thanks in advance and best regards,
Andreas

### Steps to Reproduce

see above

### Expected Behavior

see above

### Screenshots

![image](https://github.com/user-attachments/assets/4c08a02a-14fc-4d12-94b5-2be80496b208)
![image](https://github.com/user-attachments/assets/f26de4e2-5b1f-4ada-b8b5-c021fcb58c62)

### Used Versions

- Java: 17
- SAP Cloud SDK version: 5.14.0
- Spring Boot: 3.3.6

### Code Examples

```java
// Your code here
```

### Log File

<details><summary>Log file</summary>
...
</details>

### Affected Development Phase

Production

### Impact

Blocked

**Comment by CharlesDuboisSAP** [comment]
Hello @hdva2502 

I'm currently investigating the issue.
Could you please send a small example (like a unit test) that creates the HttpClient, uses it and asserts on headers?
Because I dont know how you created either `apiClient` or `execRuntime` in your screenshots.

Best regards, Charles

**Comment by hdva2502** [comment]
Hi Charles,

the sequence from Destination => openAPI-generated APIClient looks like this:

```
		HttpDestination destination = DestinationAccessor.getDestination(destinationName).asHttp();

		HttpClient httpClient = ApacheHttpClient5Accessor.getHttpClient(destination);
		HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
		clientHttpRequestFactory.setHttpClient(httpClient);

		RestTemplate restTemplate = new RestTemplate(clientHttpRequestFactory);
		restTemplate.setUriTemplateHandler(createTemplateHandler());

		ApiClient apiClient = new ApiClient(restTemplate);
		apiClient.setBasePath(destination.getUri().toString());
```
of course, wrapped by `AuthTokenAccessor.executeWithAuthToken( ... )`.

The call to execute the apiClient contains nothing special and is wrapped by resilience settings (`ResilienceDecorator.executeCallable(() -> apiClient.execute( ... ) ... `).

I'll send you the link to the git repo via teams.

Thanks,
Andreas

**Comment by CharlesDuboisSAP** [comment]
We merged the fix #662, we plan to release in 1-2 weeks

**Comment by hdva2502** [comment]
Thanks.
Do you already know the cloud sdk version that contains the fix?
Best,
Andreas

**Comment by CharlesDuboisSAP** [comment]
The fix is in 5.15.0, I will tell you when it releases

**Comment by MatKuhr** [comment]
Version `5.15.0` is available now 🥳
